core/templates/dev/head/components/question-directives/questions-list/questions-list.directive.ts

```diff
@@ -205,23 +221,32 @@ oppia.directive('questionsList', [
               ]
             });
 
-            modalInstance.result.then(function(skillId) {
-              ctrl.skillId = skillId;
-              EditableSkillBackendApiService.fetchSkill(
-                skillId).then(
-                function(skillDict) {
-                  ctrl.misconceptions = skillDict.misconceptions.map(function(
-                      misconceptionsBackendDict) {
-                    return MisconceptionObjectFactory.createFromBackendDict(
-                      misconceptionsBackendDict);
-                  });
-                  ctrl.initializeNewQuestionCreation();
-                }, function(error) {
-                  AlertsService.addWarning();
-                });
+            modalInstance.result.then(function(skillIds) {
+              ctrl.newQuestionSkillIds = skillIds;
+              ctrl.getMisconceptions(skillIds);
+              if (AlertsService.warnings.length === 0) {
+                ctrl.initializeNewQuestionCreation();
+              }
             });
           };
 
+          ctrl.getMisconceptions = function(skillIds) {
```

Comment:
Ditto here and elsewhere (re "get" misnaming)

Reply:
Done.